src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -120,9 +121,11 @@ export class TelemetryCollectionManagerPlugin
       ? this.elasticsearchClient?.asScoped(config.request).asCurrentUser
       : this.elasticsearchClient?.asInternalUser;
     // Scope the saved objects client appropriately and pass to the stats collection config
-    const soClient = config.unencrypted
-      ? this.savedObjectsService?.getScopedClient(config.request)
-      : this.savedObjectsService?.createInternalRepository();
+    const soClient =
```

Comment:
nit: this ternary / conditional is quite long, I think the code would be easier to read if we had an `if(config.unencrypted)` and then configured the es client and soClient for each of the cases. 

Thanks for the comment! Can you add that we scope it because this is a returned for a user request.

I think it's also worth adding a comment here to explain why we're returning a `TelemetrySavedObjectsClient` if it isn't encrypted (to make it clear that we're using it to collect telemetry across all spaces)

Reply:
Good point! I've refactored that part and added a few tests to make sure we are correctly picking the right client based on the `unencrypted` flag.